src/aolayer.cpp

```diff
@@ -382,10 +382,10 @@ void AOLayer::play()
     else
       this->freeze();
   }
-  else {
-      while (movie_delays.size() <= frame) {
-          frameAdded.wait(&mutex);
-      }
+  else if (!movie_delays.isEmpty()) {
```

Comment:
this is actually kind of confusing. The invariant here is that `movie_delays.size() == movie_frames.size()`, so this really means don't do this branch is there are no frames loaded yet. But `start_playback()` loaded at least one frame, so when `play()` is called, there must be at least one frame loaded unless frame loading failed. But then `max_frames` should be equal to `0` because nothing was loaded.

So I don't recommend adding this if statement.

Reply:
Okay but then the segfault comes back because the "should" doesn't happen